src/Confluent.Kafka/Admin/AlterConfigsException.cs

```diff
@@ -35,11 +35,11 @@ public class AlterConfigsException : Exception
         ///     results will be in error.
         /// </param>
         public AlterConfigsException(List<AlterConfigsExceptionResult> results)
-            : base(
+            : base(new Error(ErrorCode.Unknown,
```

Comment:
Currently, 1. a top level error will raise a `KafkaException` and 2. if one of the results is in error this will raise an `AlterConfigsException` which does not derive from `KafkaException`. It would be good if it did since there will be scenarios where handling at the `KafkaException` level would be desired.

However, there's a need for an error code for the scenario 'one of the constituents is in error'. `Unknown` is the best of the available i could see, but it's not really good enough.

If we can't get that in librdkafka pre 1.0, it would be best to revert this now (adding derivation from `KafkaException` later is a non-breaking change).

same comment on other admin client APIs.

Reply:
From a user's perspective, do I really care if the entire request failed or if one or more (possibly all) resources failed?
It is sort of pointless in first checking request-level errors and then entity-level errors, when the entity-level errors will reflect the request-level error.

Reply:
Why is the error code set to Unknown?

Reply:
most of the time you may not care, which is why you want to be able to just catch `KafkaException` (and have it work for all cases) - this is why the `Unknown` error code (which should be something else not unknown), but sometimes you do care if some of the results are in error and others aren't in which case you want to additionally catch the more specific exception.